pyro/distributions/conjugate.py

```diff
@@ -14,8 +14,17 @@ def _log_beta(x, y):
     return torch.lgamma(x) + torch.lgamma(y) - torch.lgamma(x + y)
 
 
-def _log_factorial(x):
-    return torch.lgamma(1 + x)
+def _log_beta_1(alpha, value, is_sparse):
+    if is_sparse:
+        mask = (value != 0)
+        value, alpha, mask = torch.broadcast_tensors(value, alpha, mask)
+        result = torch.zeros_like(value)
+        value = value[mask]
+        alpha = alpha[mask]
+        result[mask] = torch.lgamma(1 + value) + torch.lgamma(alpha) - torch.lgamma(value + alpha)
+        return result
+    else:
+        return torch.lgamma(1 + value) + torch.lgamma(alpha) - torch.lgamma(value + alpha)
```

Comment:
+1 I think that this method will be handy for other discrete conjugate distributions too (e.g. it can be used in BetaBinomial class).